src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -47,7 +47,8 @@ public static async Task<string> GetDigestTokenForCredential(NetworkCredential c
             // It is mandatory for servers to implement sha-256 per RFC 7616
             // Keep MD5 for backward compatibility.
             string algorithm;
-            if (digestResponse.Parameters.TryGetValue(Algorithm, out algorithm))
+            bool isAlgorithmSpecified;
+            if ((isAlgorithmSpecified = digestResponse.Parameters.TryGetValue(Algorithm, out algorithm)))
```

Comment:
can you please put the assignment before `if`
```c#
bool isAlgorithmSpecified = digestResponse.Parameters(..);
if (isAlgorithmSpecified)
{
```

Reply:
done